src/frontend/src/optimizer/rule/rewrite_like_expr_rule.rs

```diff
@@ -0,0 +1,160 @@
+                    (Some(a), Some(b)) => min(a, b),
+                    (Some(idx), None) => idx,
+                    (None, Some(idx)) => idx,
+                    (None, None) => {
+                        return FunctionCall::new_unchecked(ExprType::Equal, inputs, ret).into()
+                    }
+                };
+
+                if idx == 0 {
+                    return FunctionCall::new_unchecked(ExprType::Equal, inputs, ret).into()
+                }
+
+                let (low, high) = {
+                    let low = bytes[0..idx].to_owned();
+                    if low[idx - 1] == 255 {
+                        return FunctionCall::new_unchecked(func_type, inputs, ret).into();
+                    }
+                    let mut high = low.clone();
+                    high[idx - 1] += 1;
+                    match (from_utf8(&low), from_utf8(&high)) {
+                        (Ok(low), Ok(high)) => (low.to_owned(), high.to_owned()),
+                        _ => {
+                            return FunctionCall::new_unchecked(func_type, inputs, ret).into();
+                        }
+                    }
+                };
+
+                let between = FunctionCall::new_unchecked(
+                    ExprType::And,
+                    vec![
+                        FunctionCall::new_unchecked(
+                            ExprType::GreaterThanOrEqual,
+                            vec![
+                                ExprImpl::InputRef(x.clone()),
+                                ExprImpl::Literal(Literal::new(Some(ScalarImpl::Utf8(low.into())), DataType::Varchar).into())
+                            ],
+                            DataType::Boolean)
+                            .into(),
+                        FunctionCall::new_unchecked(
+                            ExprType::LessThanOrEqual,
+                            vec![
+                                ExprImpl::InputRef(x),
+                                ExprImpl::Literal(Literal::new(Some(ScalarImpl::Utf8(high.into())), DataType::Varchar).into())
+                            ],
+                            DataType::Boolean).into(),
+                    ],
+                    DataType::Boolean,
+                );
+
+                if idx == len - 1 {
+                    between.into()
+                } else {
+                    FunctionCall::new_unchecked(
+                        ExprType::And,
+                        vec![
+                            between.into(),
+                            FunctionCall::new_unchecked(func_type, inputs, ret).into(),
+                        ],
+                        DataType::Boolean,
+                    ).into()
+                }
+            } else {
+                FunctionCall::new_unchecked(func_type, inputs, ret).into()
+            }
+        } else {
+            FunctionCall::new_unchecked(func_type, inputs, ret).into()
+        }
```

Comment:
style: Using `let else` to put short brunches in the front might be slightly better. (and maybe return `func_call.into()`?)